src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private final int maxFigures = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(maxFigures);
+
+        String randomColor = colorSupplier.getRandomColor();
+
+        switch (randomFigure) {
+            case 0:
+                int randomSide = random.nextInt(100);
+                return new Square(randomColor, randomSide);
+            case 1:
+                int randomRadius = random.nextInt(100);
+                return new Square(randomColor, randomRadius);

```

Comment:
You are creating a square with a radius which is conceptually incorrect. Squares don't have a radius, they have sides. Perhaps you meant to create a Circle here.